5 Character Names Code/Text Support

Lorenzooone opened the pull request:
Also adds English default names.

Found all the places where there needed to be changes to multiplications by either looking myself or by using a disassembler. This means there should be no more places to change. Hopefully. In case a name isn't properly displayed (happens especially with Jeff's and Poo's), make an issue.

This commit does NOT add support at a graphical level. So there can be issues with names too long at a graphical level. (Headers call an extra tile from the checkerboard tiles if the names are too long for example)

jeffman commented:
Nice! Can you elaborate on how you reallocated the RAM? How many letters are available for the player name?

Have you tested saving and reloading battery saves with 5-letter names?

Lorenzooone commented:
I removed the japanese name, which is 0xD bytes long. It's located between Rockin and the English Player name (which still is 24 letters long and properly read thanks to this pr: https://github.com/jeffman/Mother2GbaTranslation/pull/43 ).
I moved Paula, Jeff, Poo, King, Steak and Rockin a bit. There are labels in m2-hacks which show the old position and the new position.
Basically now each character's name is 7 bytes long (5 + [00 FF]), everything is moved a bit because of that and it was possible to make it thanks to the fact that there were 0xC free bytes due to the Japanese name not being there.
This made it so there was need to change the formula which the game used to determine dynamically the characters names' positions in the RAM. Besides that, it required changing direct references, maximum length considered and numbers loaded and added in certain situations.

Yes, the game properly loads and stores 5-letter names.

(Though saves done prior to this fix have broken names now)

Lorenzooone commented:
There were also 2 free bytes between Steak and Rockin (They wanted to make the food 8 letters long...?), so there are still 0xB free bytes.
